testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java

```diff
@@ -812,14 +812,13 @@ private static StageStats getRootStage(MaterializedResult result)
         return requireNonNull(statementStats.getRootStage(), "root stage is null");
     }
 
-    private Session enableDynamicFiltering(boolean enabled)
+    protected Session enableDynamicFiltering(boolean enabled)
```

Comment:
Isn't DF enabled by default?
Why do we have this method?

Why is it also disabling CBO?

Reply:
@raunaqmorarka can you comment on this one? You introduced this test initially.

Reply:
CBO is disabled in DF tests so that the join order remains the syntactic one. We usually want to follow the test writer's intent about which table they put on probe and build for DF tests (e.g. sometime we want to test with large build tables).
DF is enabled by default, except when task retries are enabled. I added this method so that I can test joins with and without DF explicitly in the retries mode.